app/controllers/api/api_controller.rb

```diff
@@ -101,6 +97,120 @@ def dashboard
     end
   end
 
+  # Get Team & Team Role & Permissions info for the Mobile App.
+  def mobile_basic
+    @game = params[:game_id].nil? ? current_game : Game.find_by_id(params[:game_id])
+    @game.update
+
+    user_team = Team.find_by_id(current_user.team_id)
+    user_team_role = TeamRole.find_by_id(current_user.team_role_id)
+
+    begin
+      #generate overall embedded result
+      @result = {
+        'team' => user_team,
+        'team_role' => user_team_role
+      }
+    rescue
+      @status = 500
+      @message = "Failure to generate mobile basic results."
+    end
+
+
+    # if we haven't set a message, we sucessfully made stuff
+    unless @message
+      @status = 200
+      @message = "Success!"
+    end
+
+    @response = {
+      status: @status,
+      message: @message,
+      date:  Time.now,
+      result: @result
+    }
+
+    respond_to do |format|
+      format.json { render :json => @response }
+    end
+  end
+
+  # Get Public Relations / Income info for the Mobile App - Income Tab.
+  def income
+    @game = params[:game_id].nil? ? current_game : Game.find_by_id(params[:game_id])
+    @game.update
+
+    round = @game.round
+    public_relations_list = PublicRelation.where(game: @game)
+
+    begin
+      #generate overall embedded result
+      @result = {
+        "timer" => {
+          "round"=>  round,
+        },
+        "pr" => public_relations_list
+      }
+    rescue
+      @status = 500
+      @message = "Failure to generate public relation results."
+    end
+
+    # if we haven't set a message, we sucessfully made stuff
+    unless @message
+      @status = 200
+      @message = "Success!"
+    end
+
+    @response = {
+      status: @status,
+      message: @message,
+      date:  Time.now,
+      result: @result
+    }
+
+    respond_to do |format|
+      format.json { render :json => @response }
+    end
+  end
+
+ # Get Message Data for the Mobile App - Messages Tab.
+  def messages
+    @game = params[:game_id].nil? ? current_game : Game.find_by_id(params[:game_id])
+    @game.update
+
+    round = @game.round
+    messages = Message.all.order('created_at DESC')
```

Comment:
Yeah, I added the migration yesterday. 

Honestly, don't worry about the grouping or querying now. I can do it in the JS no problem, I just thought it might be easier / more efficient to do it as part of the query. We can treat it as a future refactor / improvement.